### DISCUSSION TITLE
How can I bend image like puppet warp in photoshop?

### DISCUSSION THREAD

**Discussion by tomocrafter** [ImageMagick/ImageMagick#6222] [Help]
### ImageMagick version

7.1.1-6

### Operating system, version and so on

Linux ubuntu 5.19.0-38-generic #39-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 17 17:33:16 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Description

I want to use ImageMagick to bend an image in any direction with a defined anchor point, similar to Photoshop's Puppet Warp. 
Just like this one: https://youtu.be/LdM_mVQY9W4?t=257

How can I do this?

Thanks.

**Comment by fmw42** [comment]
Imagemagick does not have any built-in puppet warp or mesh warp.

**Reply by tomocrafter** [reply]
Thank you for replying!
Yes, I also find that there is no way to do exactly same warps in Imagemagick.
But I wanted to know how we can achieve similar result with another way. currently, I am planning to do this by spliting image into 3 pieces and bend center part with Arc Distortion, and then adjust position each other along with rotating the one that needs to be connected to bended part.
I am sorry if this is hard to understand due to my English (I didn't use any translator so) or bad in explanation. I hope there is better way to achieve this!

**Comment by fmw42** [comment]
If you can do what you want in Photoshop, you might be able to create a distortion map in Photoshop copying the puppet warp and bring it back to Imagemagick for processing. If you cannot first create what you want in Photoshop, then you would have to break the image into parts, as you suggested, and process each part separately and manually align them to put them back. Alternately, you could create your own triangle mess and warp each triangle with an affine warp. See for example my bash unix Imagemagick script, messwarp, at http://www.fmwconcepts.com/imagemagick/index.php.
